db/flush_scheduler.cc

```diff
@@ -65,12 +67,12 @@ ColumnFamilyData* FlushScheduler::TakeNextColumnFamily() {
 }
 
 bool FlushScheduler::Empty() {
-#ifndef NDEBUG
-  std::lock_guard<std::mutex> lock(checking_mutex_);
-#endif  // NDEBUG
   auto rv = head_.load(std::memory_order_relaxed) == nullptr;
 #ifndef NDEBUG
-  assert(rv == checking_set_.empty());
+  std::lock_guard<std::mutex> lock(checking_mutex_);
+  // Empty is allowed to be called concurrnetly with ScheduleFlush. It would
```

Comment:
On the one hand, we call `Empty` in `PreprocessWrite` in the write thread. On the other hand, we call `ScheduleFlushes` in the write thread as well. Can they be called concurrently? Maybe I am missing some cases.

Reply:
nvm, I remember we once discussed pipelined write about this.